### PR TITLE
Typography update

### PR DESCRIPTION
**Pull request by spiegelp**
New styles for `TextBlock` and `Hyperlink` according to the new [specification](https://material.io/design/typography/the-type-system.html).

### CODE REVIEW

**Comment by jespersh** [comment]
Is that the 2.0 spec or a revision to the 1.0? 
And since I'm already asking, how do you find changes to material design spec?

**Comment by spiegelp** [comment]
@jespersh Google always made changes to the specification. However, Google introduced huge changes to Material Design and launched a new website earlier this year. As far as I know, they never called it officially Material Design 2 or used any revision numbers. Changes are only mentioned by month and year in a "What's new" section. 

Google relaunched the old website with the "old" specification last summer. You will find the old typography specification [here](https://material.io/archive/guidelines/style/typography.html).

File: MaterialDesignThemes.Wpf/Themes/MaterialDesignTheme.Hyperlink.xaml [hunk 1]
```diff
@@ -59,4 +61,76 @@
...
+                <MultiTrigger.Conditions>
+                    <Condition Property="IsEnabled" Value="True" />
+                    <Condition Property="IsMouseOver" Value="True" />
+                </MultiTrigger.Conditions>
+                <Setter Property="TextDecorations" Value="Underline" />
+            </MultiTrigger>
+        </Style.Triggers>
+    </Style>
+    <Style TargetType="{x:Type Hyperlink}" x:Key="MaterialDesign2Body2Hyperlink"
+           BasedOn="{StaticResource MaterialDesignBody1Hyperlink}">
```
Comment (Keboo) [inline MaterialDesignThemes.Wpf/Themes/MaterialDesignTheme.Hyperlink.xaml]: It looks like several of these styles are still being based off of the old hyperlink style. Is that intentional?

File: MaterialDesignThemes.Wpf/Themes/MaterialDesignTheme.TextBlock.xaml [hunk 2]
```diff
@@ -77,4 +79,95 @@
...
+        <Setter Property="FontWeight" Value="Regular" />
+    </Style>
+    <Style TargetType="{x:Type TextBlock}" x:Key="MaterialDesign2Body2TextBlock">
+        <Style.Resources>
+            <Style TargetType="Hyperlink" BasedOn="{StaticResource MaterialDesign2Body2Hyperlink}" />
+        </Style.Resources>
+        <Setter Property="FontSize" Value="14" />
+        <Setter Property="FontWeight" Value="Regular" />
+    </Style>
+    <Style TargetType="{x:Type TextBlock}" x:Key="MaterialDesign2CaptionTextBlock">
```
Comment (Keboo) [inline MaterialDesignThemes.Wpf/Themes/MaterialDesignTheme.TextBlock.xaml]: My personal opinion would be to drop the "2" from the key name on these. I do realize that there will be naming conflicts once that is done. However, some of these styles don't appear to be changing anything from the original style. This one for example appears to match the original `MaterialDesignCaptionTextBlock` unless I have missed something.

**Comment by Keboo** [comment]
For this I think we are just going to replace the existing styles, so moving to the 3.0.0 milestone since that will be a breaking change.

**Pull request merged**